core/tools.js

```diff
@@ -1991,6 +1991,30 @@
 				}
 
 				return null;
+			},
+
+			/**
+			 * Merges two objects and returns new one.
+			 *
+			 * @param {Object} obj1 A base object.
+			 * @param {Object} obj2 An object, which properties will be merged to the base one.
+			 * @returns {Object} Merged object.
+			 * @member CKEDITOR.tools.object
+			 */
+			merge: function merge( obj1, obj2 ) {
+				var copy1 = CKEDITOR.tools.clone( obj1 ),
+					copy2 = CKEDITOR.tools.clone( obj2 ),
+					tools = CKEDITOR.tools;
+
+				tools.array.forEach( tools.objectKeys( copy2 ), function( key ) {
+					if ( typeof copy2[ key ] === 'object' && typeof copy1[ key ] === 'object' ) {
+						copy1[ key ] = merge( copy1[ key ], copy2[ key ] );
```

Comment:
Can't we use simply tools.merge? `merge` is defined in a hacky way, very uncommon to our code base.